apps/app/src/components/PageEditor/HandsontableModal.tsx

```diff
@@ -96,6 +96,18 @@ export const HandsontableModal = (): JSX.Element => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpened]);
 
+  // useEffect to calculate Handsontable dimensions based on hotTableContainer and isWindowExpanded
+  useEffect(() => {
+    if (hotTableContainer != null) {
+      // Get the width and height of hotTableContainer
+      const { width, height } = hotTableContainer.getBoundingClientRect();
+
+      setHandsontableWidth(width);
+      // Set the Handsontable height based on the container height or default height if isWindowExpanded is false
+      setHandsontableHeight(!isWindowExpanded ? DEFAULT_HOT_HEIGHT : height);
+    }
+  }, [hotTableContainer, isWindowExpanded]);
```

Comment:
I would prefer not to use `useEffect` as much as possible.
Could you think of a way to implement it without using `useEffect`?

```
useEffect(() => {
    const initTableInstance = table == null ? defaultMarkdownTable : table.clone();
    setMarkdownTable(table ?? defaultMarkdownTable);
    setMarkdownTableOnInit(initTableInstance);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isOpened]);
```

as well.

Reply:
I have replaced `useEffect` with a custom method to initialize the handsontable data and calculate the width/height of the handsontable container then set the handsontable size according to the container size